src/dataprep/preprocess_utils.py

```diff
@@ -0,0 +1,118 @@
+import re
+import pandas as pd
+import numpy as np
+from typing import List, Tuple
+from sklearn.preprocessing import OrdinalEncoder, OneHotEncoder
+from sklearn.linear_model import LinearRegression
+from sklearn.experimental import enable_iterative_imputer  # IterativeImputer doesn't work without this import
+from sklearn.impute import IterativeImputer
+
+MISSINGNESS_PCT = 0.005
+
+
+def drop_small_missing_data(df: pd.DataFrame, features: List[str]) -> pd.DataFrame:
+    """
+    This method goes over the features with missing values that have very little rows with missing values and will
+    drop them.
+    E.g. dropping rows for features with less than 0.5% of missing values
+    """
+    nulls_percent = df.isna().sum() / len(df)
+    small_missing_values_features = [feature for feature in features if 0 < nulls_percent[feature] < MISSINGNESS_PCT]
+    for feature in small_missing_values_features:
+        df = df[df[feature].notnull()]
+    return df
+
+
+def feature_index_for_missing_data(df: pd.DataFrame, features: List[str]) -> pd.DataFrame:
+    """
+    This method creates a new binary column indicating which rows have missing data for each feature.
+    """
+    nulls_count = df.isna().sum()
+    for feature in features:
+        if nulls_count[feature] >= 1:
+            if feature == 'Survived':  # wouldn't happen (full data on independent variable)
+                continue
```

Comment:
Could maybe drop this feature before counting Nones as there's no reason to do it for this column in the first place